src/admin/mod.rs

```diff
@@ -30,13 +30,15 @@ use tokio::sync::oneshot;
 use pprof::protos::Message;
 use prometheus_client::encoding::text::encode;
 use prometheus_client::registry::Registry;
+use std::collections::HashMap;
 #[cfg(feature = "gperftools")]
 use tokio::fs::File;
 #[cfg(feature = "gperftools")]
 use tokio::io::AsyncReadExt;
 use tracing::{error, info};
 
 use crate::config::Config;
+use crate::proxy::Error;
```

Comment:
We shouldn't use proxy::Error here; the errors are not proxy errors. Define a new error type is preferred